### PR TITLE
feat: automatically close error buffer when process finishes

### PR DESCRIPTION
**Pull request by Fuco1**
Always create the error buffer but kill it when the process exits #173 

Also added tests that with debug they stay around.

### CODE REVIEW

**Comment by thierryvolpiatto** [comment]
Merged! Thanks.

File: async.el [hunk 1]
```diff
@@ -401,17 +401,20 @@ object will return the process object when the program is
...
          (proc (let ((process-connection-type nil))
                  (make-process
                   :name name
                   :buffer buf
                   :stderr buf-err
                   :command (cons program program-args)))))
+    (set-process-sentinel
+     (get-buffer-process buf-err)
+     (lambda (proc _change)
+       (unless (or async-debug (buffer-live-p proc))
```
Comment (basil-conto) [inline async.el:414]: Did you mean `(buffer-live-p (process-buffer proc))`?

File: tests/test-async.el [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ;; -*- lexical-binding: t -*-
+(require 'subr-x)
```
Comment (basil-conto) [inline tests/test-async.el:2]: If this is just for `string-join` then it can be wrapped in `eval-when-compile`.

**Comment by Fuco1** [comment]
Damn it I wanted `process-live-p` :D

File: tests/test-async.el (hunk 2 above)
Comment (Fuco1) [reply tests/test-async.el:2]: I don't think the tests get compiled though.

File: async.el (hunk 1 above)
Comment (Fuco1) [reply async.el:414]: Yes good catch!  I'll fix it

**Pull request merged**